lib/module.js

```diff
@@ -447,6 +447,10 @@ Module._load = function(request, parent, isMain) {
   }
 
   var module = new Module(filename, parent);
+  if (opts !== undefined && opts.dlopenFlags) {
```

Comment:
perhaps `if (opts !== null && typeof opts === 'object' && opts.dlopenFlags)` to make the check more precise.

as it is here, `require('foo', null) would throw an error